Packs/Smokescreen_IllusionBLACK/Integrations/Smokescreen_IllusionBLACK/Smokescreen_IllusionBLACK.py

```diff
@@ -0,0 +1,388 @@
+        )
+        hosts: List = response["items"]
+        for h in hosts:
+            h["services"] = ", ".join(h["services"])
+        return (
+            tableToMarkdown("IllusionBLACK Network Decoys", hosts, headerTransform=lambda s: s.capitalize()),
+            {"IllusionBlack.NetworkDecoy(val.name==obj.name)": hosts},
+            hosts
+        )
+
+    def get_ti_decoys(self):
+        """
+        Gets a list of Threat Intelligence decoys from IllusionBLACK.
+        Returns: A tuple containing the response in human readable, context data and raw response formats
+        """
+        response = self._http_request(
+            method="GET",
+            url_suffix="/decoy/recon",
+            ok_codes=(200,)
+        )
+        recon_decoys = response["items"]
+        return (
+            tableToMarkdown(
+                "IllusionBLACK TI Decoys",
+                recon_decoys,
+                headerTransform=lambda s: " ".join([w.capitalize() for w in s.split("_")])
+            ),
+            {"IllusionBlack.TIDecoy(val.name==obj.name)": recon_decoys},
+            recon_decoys
+        )
+
+    def is_host_decoy(self, host):
+        """
+        Checks if the host is an IllusionBLACK network decoy
+        Args:
+            host: The name of the entity For example: SMB-12
+        Returns: True if host is a decoy else False
+        """
+        response = self._http_request(
+            method="GET",
+            url_suffix="/decoy/hosts",
+            ok_codes=(200,)
+        )
+        hosts: List = response["items"]
+        for decoy_host in hosts:
+            if host == decoy_host["name"]:
+                return True
+        return False
+
+    def is_user_decoy(self, user):
+        """
+        Checks if the user is an IllusionBLACK AD user decoy
+        Args:
+            user: The user name of the AD user to check
+        Returns: True if user is a decoy else False
+        """
+        response = self._http_request(
+            method="GET",
+            url_suffix="/decoy/users",
+            ok_codes=(200,)
+        )
+        users: List = response["items"]
+        for decoy_user in users:
+            if user.lower() == decoy_user["user_name"]:
+                return True
+        return False
+
+    def is_subdomain_decoy(self, subdomain):
+        """
+        Checks if the subdomain is an IllusionBLACK TI decoy
+        Args:
+            subdomain: The subdomain to check. For example: experience.illusionblack.com
+        Returns: True if subdomain is a decoy else False
+        """
+        response = self._http_request(
+            method="GET",
+            url_suffix="/decoy/recon",
+            ok_codes=(200,)
+        )
+        ti_decoys: List = response["items"]
+        for ti_decoy in ti_decoys:
+            if subdomain == ti_decoy["name"]:
+                return True
+        return False
+
+    def get_events(self, limit=None, query=None, from_time=None, to_time=None):
```

Comment:
it seems that you are not exposing all the fields you send to the user, is that on purpose? Our approach says to gives the user all the information in the fetch, so he will be able to map it out himself

Reply:
As we had discussed, only 4 fields are common across the incidents. Rest other fields are dynamic. To give the user the dynamic fields a default playbook is run which updates the context data.